packages/dds/tree/src/test/feature-libraries/modular-schema/discrepancies.spec.ts

```diff
@@ -597,5 +644,66 @@ describe("Schema Discrepancies", () => {
 
 			assert.equal(isRepoSuperset(leafNodeSchema1, leafNodeSchema2), false);
 		});
+
+		describe("on field kinds for root fields of identical content", () => {
+			const allFieldKinds = Object.values(FieldKinds);
+			it("verify this test is exhaustive", () => {
+				// Test case expectations below are generated manually. When new supported field kinds are added, this suite must be updated.
+				// This likely also necessitates changes to the production code this describe block validates.
+				assert.equal(allFieldKinds.length, 5);
```

Comment:
I love that you're thinking ahead to prevent rot. I do worry that the `testCases` array could end up with missing entries though, so perhaps `assert.equal(testCases.lenght, allFieldKinds.length * allFieldKinds.length)` would be safer.
That doesn't check for duplicates though. Up to you how far you want to push it.